interval/search_tree.go

```diff
@@ -113,6 +116,46 @@ func (st *SearchTree[V, T]) IsEmpty() bool {
 	return st.root == nil
 }
 
+// GobEncode encodes the tree (compatible with encoding/gob).
+func (st *SearchTree[V, T]) GobEncode() ([]byte, error) {
+	var b bytes.Buffer
+	enc := gob.NewEncoder(&b)
+
+	if err := enc.Encode(st.config.allowIntervalPoint); err != nil {
+		return nil, err
+	}
+
+	if st.root != nil {
+		if err := enc.Encode(st.root); err != nil {
+			return nil, err
+		}
+	}
+
+	return b.Bytes(), nil
+}
+
+// GobDecode decodes the tree (compatible with encoding/gob).
```

Comment:
Same as above.
```suggestion
// GobDecode decodes the tree (compatible with [encoding/gob]).
```

Reply:
Done: [Add links to [encoding/gob]](https://github.com/rdleal/intervalst/pull/9/commits/fa748e2e243ac7e6142d7ebb8b43f9dc46f8dd71)